src/Middleware/RequestThrottling/src/RequestThrottlingMiddleware.cs

```diff
@@ -46,11 +47,21 @@ public RequestThrottlingMiddleware(RequestDelegate next, ILoggerFactory loggerFa
         /// <returns>A <see cref="Task"/> that completes when the request leaves.</returns>
         public async Task Invoke(HttpContext context)
         {
-            var waitInQueueTask = _requestQueue.EnterQueue();
+            var waitInQueueTask = _requestQueue.TryEnterQueueAsync();
 
             if (waitInQueueTask.IsCompletedSuccessfully)
             {
-                RequestThrottlingLog.RequestRunImmediately(_logger);
+                var requestPassedThrough = await waitInQueueTask;
+                if (requestPassedThrough)
+                {
+                    RequestThrottlingLog.RequestRunImmediately(_logger);
+                }
+                else
+                {
+                    // the queue is full
+                    context.Response.StatusCode = StatusCodes.Status503ServiceUnavailable;
```

Comment:
This should definitely be logged.

Reply:
This should be a 429 or something

Reply:
ASP.NET returned 503.2. I don't think 429 is correct because it isn't an individual client sending too many requests.

Reply:
Agreed, a 429 would unfairly blame the client. 503 is right for a server issue.
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status
429 Too Many Requests
The user has sent too many requests in a given amount of time ("rate limiting").
503 Service Unavailable
The server is not ready to handle the request. Common causes are a server that is down for maintenance or that is overloaded.

Reply:
Yeah. @Tratcher and I were discussing this earlier. 503 is definitely the right status code, but eventually we can add an option to specify a custom request queue full RequestDelegate that could set a Retry-After header, change the status code to something else or whatever.

Reply:
Yeah, `Task OnRejected(HttpContext)` where you can provide your own response.